Validation/HGCalTriggerPrimitives/plugins/HGCalTriggerValidator.cc

```diff
@@ -0,0 +1,281 @@
+// system include files
+#include <memory>
+#include <iostream>
+#include <string>
+#include <vector>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Common/interface/Provenance.h"
+
+#include <DQMServices/Core/interface/DQMStore.h>
+#include <DQMServices/Core/interface/MonitorElement.h>
+#include <DQMServices/Core/interface/DQMEDAnalyzer.h>
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalTower.h"
+
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+
+#include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterIdentificationBase.h"
+
+//
+// class declaration
+//
+
+class HGCalTriggerValidator : public DQMEDAnalyzer {
+
+  public:
+    explicit HGCalTriggerValidator(const edm::ParameterSet&);
+    ~HGCalTriggerValidator() override;
+
+    void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  protected:
+    void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+
+  private:
+// ----------member data ---------------------------
+    //histogram tc related
+    MonitorElement *h_tc_n_, *h_tc_layer_, *h_tc_mipPt_;
```

Comment:
Would be clearer to have only one line per element. 
They can eventually be grouped, but only for variables which are connected (e.g. eta/phi,  x/y/z, etc.). Currently the grouping looks a bit random.